### ISSUE TITLE
Transmission Unstable under heavy load

### ISSUE THREAD

**Issue by FireCulex** [openwrt/packages#574]
If you add a torrent and it starts to download transmission quits. nothing in any error log. if it reaches 2.2mb/s it quits. 4-5 torrents at the same time = quits. have to manually restart every day cause startup enabled doesnt mean restart if crashed...

wndr3800 r41988 seen this atleast in the past 2000 revisions as well

I've just restarted the daemon 6 times in the past 5 minutes. Again nothing in any system logs.

Other than the billion OpenWrt transmission-daemon[2447]: Couldn't connect socket 33 to xxxx:x:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx, port xxxxx (errno 1 - Operation not permitted) (net.c:288) I get every day.

transmission-gui just becomes nonresponsive. Connection timed out.

Mem: 108128K used, 18264K free, 0K shrd, 468K buff, 14208K cached
CPU:  16% usr  42% sys   0% nic   0% idle  22% io   0% irq  19% sirq
Load average: 5.19 4.32 4.16 2/94 4139
  PID  PPID USER     STAT   VSZ %VSZ %CPU COMMAND
 4116     1 root     S    36824  29%  23% /usr/bin/transmission-daemon -g /medi
 2504     2 root     DW<      0   0%  10% [loop3]
 2497     2 root     DW<      0   0%  10% [loop2]
 2494     2 root     SW<      0   0%  10% [loop1]
 2488     2 root     RW       0   0%   5% [cifsd]
 2470     2 root     SW       0   0%   5% [cifsd]
 2476     2 root     SW       0   0%   3% [cifsd]
 2692     2 root     SW       0   0%   2% [kworker/0:1]
 2520     2 root     DW       0   0%   2% [md0_raid5]
 1823     2 root     SW       0   0%   2% [kworker/u2:4]
 1620     1 root     S     3200   3%   1% /usr/sbin/nmbd -D

Load reaches 6 transmission dies. Or maybe 89% io. Or 0k free memory? Difficult to tell whats going on.

Changed the cache size to 2mb and now the router is unresponsive. Infact it rebooted by itself after a 17 day uptime...

**Comment by ryzhovau** [comment]
Please, try to run `echo 16384 > /proc/sys/vm/min_free_kbytes` before starting transmission.

**Comment by FireCulex** [comment]
#cat min_free_kbyte
1420

is the default. changing it to 16384 appears to have more free memory. So I
overloaded it with 4 torrents and 1200kb/s bandwidth, after about 10
minutes the router stopped responding completely. Had to do a hard reset.

Mem: 102096K used, 24296K free, 0K shrd, 512K buff, 15396K cached
CPU:  32% usr  41% sys   0% nic   0% idle  10% io   0% irq  15% sirq
Load average: 3.34 3.12 1.73 2/100 25546
  PID  PPID USER     STAT   VSZ %VSZ %CPU COMMAND
25520     1 root     S    35096  28%  39% /usr/bin/transmission-daemon -g
/medi

**Comment by nmav** [comment]
My issue was also with transmission taking all the memory and oob killing arbitrary processes. My solution was to disable overcommit from the kernel:
echo 2 > /proc/sys/vm/overcommit_memory       
echo 60 >/proc/sys/vm/overcommit_ratio

(the latter value may vary on your system, but 50 or 60 should be ok)
That kills transmission once it takes more memory than this percentage, hence allowing the router to operate.

**Comment by FireCulex** [comment]
well, I switched to 3 queue size with unlimited bandwidth and it appears to
work with min_free_kybtes. Few days should confirm if this works.

My problem originally really wasn't so much as the router processes
stopping but transmission terminating within 5 minutes and not knowing till
hours later as it did not restart.

**Comment by FireCulex** [comment]
Tried again today, after 5 minutes of starting transfers the router is no
longer accessible. Routing appears to be ok as I'm using it wirelessly,
however the transmission client, the web server, the samba server, and the
ssh server have all died. min_free_kybtes appears to make the situation
worse as before, only the transmission client would terminate usually and
in rare cases the samba and ssh server, but the web server would be
accessible.  Personally I think the fact that the ssh server and or web
server doesn't restart on it's own, is in my opinion, a fatal design flaw
of OpenWRT.

**Comment by nmav** [comment]
I doubt min_free_kbytes can help. It is made for a different reason and the fact that it works for some is only incidental. If you want to prevent the system from crashing use the overcommit disable I list above. Transmission not restarting on crash is another issue. It is better to use another bug for that.

**Comment by FireCulex** [comment]
well as i said before, i'd have to restart transmission about 10-20 times
before it brought down other processes which i could restart using ssh or
the web just fine. So disabling overcommit really doesn't do anything for
the original problem which was transmission constantly crashing without
error nor reason given within 5 minutes of transfering. Don't know if it's
a load issue related to bandwidth as I see load averages spike up to 7. At
1.5mb/s it's about 6. Is it the kernel doing it I dont know, theres no log
anywhere that says anything.

**Comment by FireCulex** [comment]
The time it took me to write down the last message, the router crashed and
required a hard reboot after running transmission for 5 minutes.

**Comment by FireCulex** [comment]
disabling overcommit didnt work

**Comment by nmav** [comment]
Disabling overcommit does require a percentage of memory to use. The selection of it depends on the available memory of your router and the number of processes you run. What percentage did you use?

**Comment by nmav** [comment]
Patch #1363 switches transmission init file to procd and sets resource limits for it, in addition to asking procd to restart it.

**Comment by FireCulex** [comment]
I did echo 2 > /proc/sys/vm/overcommit_memory and echo 60
>/proc/sys/vm/overcommit_ratio
but then it caused samba to create multiple D processes and a load of 12.

So scrapping that I limited transmission to 300kb/s download and its
working like a charm. For days!

**Comment by nmav** [comment]
That's very nice for you FireCulex but I doubt we can generalize your solution for other users of openwrt. The goal of this thread is to fix it for everyone, not just your setup.

**Comment by FireCulex** [comment]
I'm just saying in my particular case, transmission was killing the router
and/or processes due to not having any bandwidth restriction. At about
900k+ or so it starts killing itself, at about 1.2mb/s other processes
start getting killed along with transmission, and about 1.5mb+ the router
overloads requiring a hard reset. There's nothing being said anywhere in
/var/log about whats going on but its clear that it has no problem with
bandwidth being restricted. The default setup of unrestricted bandwidth in
embedded systems may not be the correct solution? Maybe not try to download
4 at a time? idk..

**Comment by FireCulex** [comment]
prefetch-enabled was enabled despite being a lightweight build. This
was causing 8+ load bringing down the router at high speeds.

**Comment by diizzyy** [comment]
@hnyman 
Can you close this?
